src/SignalR/common/Shared/AsyncEnumerableAdapters.cs

```diff
@@ -51,37 +51,10 @@ public IAsyncEnumerator<TResult> GetAsyncEnumerator(CancellationToken cancellati
                     {
```

Comment:
`registration` not referenced

Reply:
The class is still needed because we need to be able to cancel the inner stream operation from the token passed into `GetASyncEnumerator` but you bring up a good point about the registration. It should be disposed